docs/units/type_hints.rst

```diff
@@ -0,0 +1,136 @@
+Unit-Aware Type Annotations
+***************************
+
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+
+While Python is a dynamically typed language, there is also support for type hints (and even static typing) using the annotations syntax of `PEP 484 <https://www.python.org/dev/peps/pep-0484/>`_.
+For a detailed guide on type hints, function annotations, and other related syntax see the `Real Python Guide <https://realpython.com/python-type-checking/#type-aliases>`_. 
+Here we how Astropy |Quantity| can be used in type hints and annotations and also include metadata about the associated units.
```

Comment:
Maybe: "Below, we describe how you can use ..."

Reply:
✅